engine/src/main/java/io/zeebe/engine/processing/deployment/distribute/CompleteDeploymentDistributionProcessor.java

```diff
@@ -41,8 +46,13 @@ public void processRecord(
     stateWriter.appendFollowUpEvent(
         deploymentKey, DeploymentDistributionIntent.COMPLETED, record.getValue());
 
-    // todo(zell): https://github.com/zeebe-io/zeebe/issues/6173
-    // check whether it was the last pending
-    // then write Deployment.FULLY_DISTRIBUTED
+    if (!deploymentState.hasPendingDeploymentDistribution(deploymentKey)) {
+      final var storedDeploymentRecord = deploymentState.getStoredDeploymentRecord(deploymentKey);
+
+      if (storedDeploymentRecord != null) {
+        stateWriter.appendFollowUpEvent(
+            deploymentKey, DeploymentIntent.FULLY_DISTRIBUTED, storedDeploymentRecord);
+      }
```

Comment:
(optional) We don't need to load the deployment record from the state. Same as in the other processor, we could just write an empty record.

Reply:
Agreed is consistent then.